ethstats/ethstats.go

```diff
@@ -537,10 +537,16 @@ func (s *Service) reportHistory(conn *websocket.Conn, list []uint64) error {
 	// Gather the batch of blocks to report
 	history := make([]*blockStats, len(indexes))
 	for i, number := range indexes {
+		var block *types.Block
 		if s.eth != nil {
-			history[len(history)-1-i] = s.assembleBlockStats(s.eth.BlockChain().GetBlockByNumber(number))
+			block = s.eth.BlockChain().GetBlockByNumber(number)
 		} else {
-			history[len(history)-1-i] = s.assembleBlockStats(types.NewBlockWithHeader(s.les.BlockChain().GetHeaderByNumber(number)))
+			if header := s.les.BlockChain().GetHeaderByNumber(number); header != nil {
+				block = types.NewBlockWithHeader(header)
+			}
+		}
+		if block != nil {
+			history[len(history)-1-i] = s.assembleBlockStats(block)
```

Comment:
Can't we just let `assembleBlockStats` return nil when the given block is nil instead of grabbing internally the head block? Basically if you don't give it anything you don't get anything. I would prefer that over an external check if this function should be called. It would mean that the code to fetch the head in `assembleBlockStats` must be moved to `reportBlock` or into a separate function that is called from `reportBlock`.